src/common/transformations/src/transformations/symbolic_transformations/symbol_optimization.cpp

```diff
@@ -402,27 +428,75 @@ struct OutputValue {
+    if (output.get_tensor().get_value_symbol().size() < 2)
+        return;  // singular values are handled by optimize_single_value_usage helper
+    const auto result = OutputValue::make(output);
+    if (!result)
+        return;
+    if (multi_symbol_source.count(*result)) {
+        // multiple value source have been seen before
+        auto alternative_source = multi_symbol_source[*result];
+        if (output.get_element_type() != alternative_source.get_element_type()) {
+            auto convert = std::make_shared<ov::op::v0::Convert>(alternative_source, output.get_element_type());
+            // should this be a convert like? no, output should be erased ideally. or should we just avoid dynamic data
+            // type?
+            ov::copy_runtime_info(output.get_node_shared_ptr(), convert);
+            alternative_source = convert->output(0);
+        }
+        if (output.get_partial_shape() != alternative_source.get_partial_shape()) {
+            const auto& shape = ov::op::v0::Constant::create(ov::element::i32,
+                                                             ov::Shape{output.get_shape().size()},
+                                                             output.get_shape());
+            alternative_source = std::make_shared<ov::op::v1::Reshape>(alternative_source, shape, false)->output(0);
+        }
+        output.replace(alternative_source);
+    } else {
+        // new instance of multiple value source
+        ov::OutputVector to_be_concated;
+        for (const auto& el : result->value) {
+            if (el.is<int64_t>()) {
+                const auto& value = el.as<int64_t>();
+                const auto& constant = ov::op::v0::Constant::create(output.get_element_type(), ov::Shape{1}, {value});
+                to_be_concated.push_back(constant->output(0));
+            } else if (el.is<std::shared_ptr<ov::Symbol>>()) {
+                const auto& symbol = el.as<std::shared_ptr<ov::Symbol>>();
+                auto alternative_output =
+                    get_alternative_source_from_value_or_shape_source(symbol_shape_source,
+                                                                      symbol,
+                                                                      ov::Shape{1},
+                                                                      output.get_element_type(),
+                                                                      output.get_node_shared_ptr(),
+                                                                      symbol_value_source);
+                if (alternative_output.get_node_shared_ptr())
+                    to_be_concated.push_back(alternative_output);
+                else
+                    break;
             } else {
-                multi_symbol_source[*result] = output;
+                break;
+            }
+        }
+        if (to_be_concated.size() != ov::shape_size(output.get_shape())) {
```

Comment:
should we add a dynamic rank check here?

Reply:
not really, we will exit this function via `if (output.get_tensor().get_value_symbol().size() < 2)` or `if (!result)` earlier.